cisco_ise/actions/lib/base.py

```diff
@@ -0,0 +1,469 @@
+from st2common.runners.base_action import Action
+from requests.auth import HTTPBasicAuth
+
+import sys
+import requests
+import json
+
+# Constants
+HTTPS_PORT = 9060
+HTTP_PORT = 80
+
+HTTP_METHODS = ["GET", "POST", "PUT", "DELETE"]
+
+#                   key, required, default
+CONNECTION_FIELDS = [
+    ("api_user", True, None),
+    ("api_pass", True, None),
+    ("hostname", True, None),
+    ("verify_ssl", True, True),
+    ("use_http", True, False),
+]
+
+MAX_PAGE_SIZE = 100
+
+#                           http_code, message
+HTTP_STATUS_RESPONSE_MESSAGE = [
+    (201, "Resource was successfully created"),
+    (204, "Resource was successfully deleted"),
+]
+
+
+class BaseAction(Action):
+    """The base class for all actions"""
+
+    def __init__(self, config):
+        """init method, run at class creation"""
+        super(BaseAction, self).__init__(config)
+        self.logger.debug("Instantiating BaseAction()")
+        self.action_kwargs = None
+        self.connection = None
```

Comment:
```suggestion
        self.connection = None
        self.payload_format = None
```
And the other properties added in `_prepare_kwargs`
Generally I like to see the attributes in `__init__` or as class-level attributes before using/modifying them in methods like `_prepare_kwargs`. this looks like a decent explanation: https://stackoverflow.com/a/19292653

Reply:
If I understand...

Methods used to breakup init is fine. Changing attributes in one of these init setup methods is fine. Creating the attribute in the init sub method... not desirable. 

We want to always define all attributes at the head of init, even if we immediately change it in the init logic.

If I have that correct then I am 100% on board and that is a superior way of handling it.

Reply:
I have added all attributes to the main `__init__()` method as `None` and added a comment. Good suggestion, thank you!